build.gradle

```diff
@@ -61,6 +61,7 @@ subprojects {
   detekt {
     config = files("$rootDir/.detekt.yml", "$rootDir/.detekt-compose.yml")
     buildUponDefaultConfig = true
+    source = files("src", "$rootDir/buildSrc/src")
```

Comment:
By default, "detekt" task only runs for src/main/java, src/test/java, src/main/kotlin and src/test/kotlin. See the docs here: https://detekt.dev/docs/gettingstarted/gradle

By using "src", we always run detekt regardless tests/main, flavors, etc.

There's "detektTest" task, but it doesn't support auto-correct. Also, it increases the build time to run 2 tasks instead of one.

Also, buildSrc is a separate project, and all we set up in the root build.gradle is not applicable there.
Therefore, in order to run detekt there, you need to set up the same stuff in buildSrc/build.gradle, and run it like this: "./gradlew -p buildSrc :detekt".

Or, since detekt is a static analyser, it's enough just to include "buildSrc/src" :)

The only potential downside is that the check will be run for each module, but it's quite small and doesn't increase the check time to any noticeable way.